### PR TITLE
WIP: Docker 1.12

### PR DESCRIPTION
**Pull request by mischief**
this PR is currently broken because docker 1.12.0 needs runc 1.0.0, for which no non-rc tag exists yet.

also depends on #2123

fixes coreos/bugs#1501

### CODE REVIEW

**Comment by mischief** [comment]
depends on coreos/docker#25
will need to fix commit hash in the docker ebuild once that lands

File: app-emulation/docker/docker-9999.ebuild [hunk 1]
```diff
@@ -66,46 +66,46 @@ RDEPEND="
...
 	>=app-arch/xz-utils-4.9
 
-	>=app-emulation/containerd-0.2.0
-	>=app-emulation/runc-0.1.0
-
-	apparmor? (
-		sys-libs/libapparmor[static-libs]
-	)
+	>app-emulation/containerd-0.2.2
+	app-emulation/runc[apparmor?,seccomp?]
```
Comment (marineam) [inline app-emulation/docker/docker-9999.ebuild]: Should this specify version?

Comment (mischief) [inline app-emulation/docker/docker-9999.ebuild]: probably! :-)

File: app-emulation/docker/files/docker.service [hunk 2]
```diff
@@ -5,11 +5,27 @@ After=docker.socket early-docker.target network.target
...
+LimitNPROC=infinity
+LimitCORE=infinity
+# Uncomment TasksMax if your systemd version supports it.
+# Only systemd 226 and above support this version.
+TasksMax=infinity
+TimeoutStartSec=0
+# set delegate yes so that systemd does not reset the cgroups of docker containers
+Delegate=yes
+# kill only the docker process, not all processes in the cgroup
+KillMode=process
```
Comment (marineam) [inline app-emulation/docker/files/docker.service]: Ick, I know this is what upstream did but if you want to keep containerd running please move it to a new service so it is possible to restart it.

File: app-emulation/docker/files/early-docker.service [hunk 3]
```diff
@@ -5,11 +5,27 @@ After=early-docker.socket
...
+LimitNPROC=infinity
+LimitCORE=infinity
+# Uncomment TasksMax if your systemd version supports it.
+# Only systemd 226 and above support this version.
+TasksMax=infinity
+TimeoutStartSec=0
+# set delegate yes so that systemd does not reset the cgroups of docker containers
+Delegate=yes
+# kill only the docker process, not all processes in the cgroup
+KillMode=process
```
Comment (marineam) [inline app-emulation/docker/files/early-docker.service]: Ditto

**Comment by mischief** [comment]
PTAL, dropped `KillMode=process`, fixed DEPEND version/use, bumped to docker 1.12.1.

**Comment by marineam** [comment]
LGTM, at some point here we really should run containerd as a proper service but status quo is fine for now.

**Pull request merged**